### PR TITLE
proxy: Use fixed delay for proxy connection

### PR DESCRIPTION
**Pull request by bpradipt**
Fixes: #1012

### CODE REVIEW

File: pkg/adaptor/proxy/proxy.go [hunk 1]
```diff
@@ -126,6 +126,7 @@ func (p *agentProxy) dial(ctx context.Context, address string) (net.Conn, error)
 		},
 		retry.Attempts(0),
 		retry.Context(ctx),
+		retry.DelayType(retry.FixedDelay),
```
Comment (yoheiueda) [inline pkg/adaptor/proxy/proxy.go:129]: The default delay is 100ms, and I think it causes too frequent retries.

We can set an upper limit for default backoff delays like this.
https://github.com/confidential-containers/cloud-api-adaptor/blob/ade303e0acddcae9aa94bc753866578215587ae7/pkg/adaptor/cloud/ibmcloud-powervs/provider.go#L107-L109

Comment (bpradipt) [reply pkg/adaptor/proxy/proxy.go:129]: Thanks @yoheiueda. I'll try setting MaxDelay and report the results

**Comment by bpradipt** [comment]
This is the o/p when using MaxDelay. Approx time is 1 min 
15:17:34 -> Created VM successfully
15:18:23 -> agent proxy ready

```
2023/05/30 15:17:34 [adaptor/cloud/azure] created VM successfully: /subscriptions/89a9ba4b-8b66-446a-813d-a9ed3a129e3d/resourceGroups/bpradipt-pp1-rqf7p-rg/providers/Microsoft.Compute/virtualMachines/podvm-network-debug-app-f67f86b75-q7sn8-2e342077
2023/05/30 15:17:34 [adaptor/cloud/azure] podNodeIP[0]=10.0.128.10
2023/05/30 15:17:34 [util/k8sops] network-debug-app-f67f86b75-q7sn8 is now owning a PeerPod object
2023/05/30 15:17:34 [adaptor/cloud] created an instance podvm-network-debug-app-f67f86b75-q7sn8-2e342077 for sandbox 2e342077c0615c6bfa28924177fb508b8eed39526eccb0a2bc14d0d3f3b57c95
2023/05/30 15:17:34 [tunneler/vxlan] vxlan ppvxlan1 (remote 10.0.128.10:9000, id: 555000) created at /proc/1/task/10/ns/net
2023/05/30 15:17:34 [tunneler/vxlan] vxlan ppvxlan1 created at /proc/1/task/10/ns/net
2023/05/30 15:17:34 [tunneler/vxlan] vxlan ppvxlan1 is moved to /var/run/netns/57242460-033b-4340-8de6-5af82748fffd
2023/05/30 15:17:34 [tunneler/vxlan] Add tc redirect filters between eth0 and vxlan1 on pod network namespace /var/run/netns/57242460-033b-4340-8de6-5af82748fffd
2023/05/30 15:17:34 [adaptor/proxy] Listening on /run/peerpod/pods/2e342077c0615c6bfa28924177fb508b8eed39526eccb0a2bc14d0d3f3b57c95/agent.ttrpc
2023/05/30 15:17:34 [adaptor/proxy] failed to init cri client, the err: cri runtime endpoint is not specified, it is used to get the image name from image digest
2023/05/30 15:17:34 [adaptor/proxy] Trying to establish agent proxy connection to 10.0.128.10:15150
2023/05/30 15:18:23 [adaptor/proxy] established agent proxy connection to 10.0.128.10:15150
2023/05/30 15:18:23 [adaptor/cloud] agent proxy is ready
```

**Comment by surajssd** [comment]
Is the maxdelay slower? If the same machine image was used?

**Review by yoheiueda — APPROVED** [review]
LGTM

**Review by snir911 — APPROVED** [review]
LGTM thanks

**Pull request merged**